### PR TITLE
Internal tooling updates

### PR DESCRIPTION
**Pull request by tysongach**
Some general housekeeping for internal tooling and reducing warning about vulnerabilities in our npm sub-dependencies used to run the contributing page. **CircleCI will fail until #649 is merged.**

### CODE REVIEW

**Comment by whmii** [comment]
The gulp changes seem gravy to me. I won't have an opportunity to do a test run today, but it reads fine to me. I'm not sure why hound is upset since, while `const` is a keyword, the syntax seems correct.

File: Gulpfile.js [hunk 1]
```diff
@@ -1,33 +1,42 @@
...
-  return gulp.src(paths.scss)
-    .pipe(sass({
-        sourcemaps: true,
-        precision: 6
-    }))
-    .pipe(autoprefix("last 2 versions"))
-    .pipe(gulp.dest("./contrib"))
-    .pipe(connect.reload());
-});
+function styles() {
```
Comment (whmii) [inline Gulpfile.js]: Why are we changing the syntax of all of these tasks? This isn't typically the way one would author a guplfile.

File: Gulpfile.js [hunk 2]
```diff
@@ -1,33 +1,42 @@
-var autoprefix = require("gulp-autoprefixer"),
```
Comment (whmii) [inline Gulpfile.js]: Upon contemplating this, I support this change as it is more true to the actual output of the library.

**Comment by whmii** [comment]
Some of the changes to they syntax of the gulpfile are perplexing to me and seem to be copy/pasted from somewhere else.

Broadly speaking it no longer 'looks' like a gulp file. I can adjust some of these changes to use more regular gulp promises if there's interest, but to move to lots of open functions seems like a lateral/backwards move.

**Comment by tysongach** [comment]
@whmii 👇 

> Seems like the ruby version is tripping up CI. is this due to .tool-versions? Ive never seen that file type before.

Yeah, I mentioned up in the PR description that Circle will fail until #649 is merged. `.tool-versions` comes from [asdf](https://github.com/asdf-vm/asdf), which is now our preferred programming language version manager and part of the [laptop script](https://github.com/thoughtbot/laptop).

> The browser-sync changes seem gravy to me.

I agree, but gulp-connect seems to not get updated much anymore and was causing security vulnerabilities via its dependencies. I specifically started this work to cut those vulnerabilities out, based on a comment in the `#design` channel.

File: Gulpfile.js (hunk 1 above)
Comment (tysongach) [reply Gulpfile.js]: This is a feature of gulp v4 and appears to be the recommended way. I followed the gulp docs recipe to get to this point: https://github.com/gulpjs/gulp/blob/master/docs/recipes/minimal-browsersync-setup-with-gulp4.md

**Comment by whmii** [comment]
LGTM @tysongach

**Comment by tysongach** [comment]
Thanks, @whmii

**Pull request merged**